### ISSUE TITLE
Block device sequential write performance fluctuates widely

### ISSUE THREAD

**Issue by maydaycc** [cloud-hypervisor/cloud-hypervisor#5164]
**Describe the bug**
Block device sequential write performance fluctuates widely

**To Reproduce**
Steps to reproduce the behaviour:
Repeat fio seq-write-test in guest OS，the results may be different each time, but the performance of a single time is relatively stable.

Observe the disk statistics through iostat on the host. It seems that the request size is a random value and has nothing to do with fio -bs. The higher the value, the better the test result.

The same test is performed on a qemu VM, the request size is consistent with the fio -bs, and the results of multiple tests are also close.

fio -direct=1 -iodepth=32 -rw=write -ioengine=libaio -bs=1024k -numjobs=1 -runtime=300 -group_reporting -filename=/dev/vdb -name=Write_BandWidth_Test --time_based

TEST1
![image](https://user-images.githubusercontent.com/26605513/215917639-11db90cf-fbac-4c21-a083-5d72b959a43b.png)

![image](https://user-images.githubusercontent.com/26605513/215917314-d569b793-c9d6-4894-80b6-ad3364d1c42f.png)

TEST2
![image](https://user-images.githubusercontent.com/26605513/215918608-3bfe3559-2bfc-4c33-b06b-914fc39f2c79.png)

![image](https://user-images.githubusercontent.com/26605513/215917725-ccd04757-005d-4956-a2f1-7cdd8674b3dd.png)

**Version**
v28.0.0

**VM configuration**

./cloud-hypervisor-scsi \
        --cpus boot=4 \
        --memory size=4096M \
        --kernel vmlinux-5.10.25-88 \
        --cmdline "console=ttyS0 root=/dev/vda1 rw iommu=off" \
        --disk path=focal-server-cloudimg-amd64-2.raw path=/dev/rbd0,direct=on,num_queues=4,queue_size=1024 \
        --console off \
        --serial tty 

Guest OS version details:
NAME="Ubuntu"
VERSION="20.04.5 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.5 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal

Host OS version details:
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

**Comment by maydaycc** [comment]
> How are you accounting for the fact that your RBD device could be accounting for that variability?

Attach the same RBD device to a qemu VM, results on the qemu VM are very stable。

**Comment by maydaycc** [comment]
> How are you accounting for the fact that your RBD device could be accounting for that variability?
Attach a local nvme SSD disk to a CLH vm 
![image](https://user-images.githubusercontent.com/26605513/216198306-97c8e4e1-b436-4f76-8529-1f26d5fb4108.png)

![image](https://user-images.githubusercontent.com/26605513/216198595-974859b4-2b6c-4d2a-829d-0784a8fecc31.png)

![image](https://user-images.githubusercontent.com/26605513/216198666-9d7fd45f-e1b1-477b-bb22-d967505649e0.png)

**Comment by maydaycc** [comment]
I wonder if it related to the VIRTIO_ BLK_ F_ SEG_ MAX feature?

CLH VM:
![image](https://user-images.githubusercontent.com/26605513/216200381-7ef0585e-a649-4e07-89c6-abf7c2cd91fe.png)
QEMU VM:
![image](https://user-images.githubusercontent.com/26605513/216200411-8016b544-a751-4f14-8ca4-e42830b85183.png)

**Comment by likebreath** [comment]
@maydaycc You are right that the `VIRTIO_BLK_F_SEG_MAX` feature is not implemented in our virtio-blk device. Not sure if this is what makes a difference though. Contributions are welcome to add this feature support. Also, you may want to give it a try for `vhost-user-blk` (which seems to enable this particular feature).

@sboeuf is our virtio expert. Any comments?

**Comment by liuw** [comment]
I have a hunch that if the max segment becomes configurable we need to enlarge the default SmallVec size used in the block processing code. That itself is not a big deal, but probably something easy to miss until we run dhat profiling again.
